### PR TITLE
support future long running tasks

### PR DESCRIPTION
**Pull request by kyryloonufriiev**
This PR is changing the way how `ServerConnector` handle long requests related to changes in Eyes server version 10.9

https://trello.com/c/60Rm4xXG/240-support-future-long-running-tasks

### CODE REVIEW

**Comment by amitzur** [comment]
@tourdedave and @danielschwartz85 - please review this PR as well as the trello card. It's a bit long read, but it's a small code change and would be good for you to be acquainted with the protocol. Add your review to this PR.
@kyryloonufriiev - once this is approved, merge to master (make sure tests pass in at least eyes-sdk-core and eyes-selenium, and add a changelog entry for eyes-sdk-core).

Theoretically we should release all SDK's as a result, but there's no rush and I think we can batch it with other changes next week. WDYT (everyone)?

File: packages/eyes-sdk-core/lib/server/requestHelpers.js [hunk 1]
```diff
@@ -108,70 +110,64 @@ async function delayRequest({axiosConfig, logger}) {
...
+    delay: TypeUtils.isArray(config.delayBeforePolling)
+      ? config.delayBeforePolling[0]
+      : config.delayBeforePolling,
+    method: 'GET',
+    url,
+  }
+  const response = await axios.request(pollingConfig)
+  switch (response.status) {
+    case HTTP_STATUS_CODES.OK:
+      return response
```
Comment (tourdedave) [inline packages/eyes-sdk-core/lib/server/requestHelpers.js]: Based on what I read in the protocol specified in the Trello ticket -- if we're polling and the response is `200` we should make another call to continue polling. But this looks like we're just returning the response. Am I missing something?

Comment (kyryloonufriiev) [reply packages/eyes-sdk-core/lib/server/requestHelpers.js]: I think you missed the place where this logic is handled, look at `handleRequestResponce` line 124. Line 150 will not be resolved until polling is finished (return non-200).

Comment (tourdedave) [reply packages/eyes-sdk-core/lib/server/requestHelpers.js]: Ah, I see. This function is called from the `axios` receptor. That makes more sense.

File: packages/eyes-sdk-core/test/unit/server/ServerConnector.spec.js [hunk 2]
```diff
@@ -103,6 +103,7 @@ describe('ServerConnector', () => {
       const response = {status: 200, config, data: {}, headers: {}, request: {}}
       if (config.isLongRequest) {
```
Comment (amitzur) [inline packages/eyes-sdk-core/test/unit/server/ServerConnector.spec.js]: I'd like to see a test that shows that the client initiates a request without the `isLongRequest` flag, the server responds with 202, and then we assert that long polling took place.

Comment (kyryloonufriiev) [reply packages/eyes-sdk-core/test/unit/server/ServerConnector.spec.js]: I add a specific test to check all nuances of polling protocol. Can I merge it?

**Pull request merged**